sql/api/src/main/scala/org/apache/spark/sql/streaming/StatefulProcessorHandle.scala

```diff
@@ -30,58 +30,72 @@ import org.apache.spark.sql.Encoder
 private[sql] trait StatefulProcessorHandle extends Serializable {
 
   /**
-   * Function to create new or return existing single value state variable of given type. The user
-   * must ensure to call this function only within the `init()` method of the StatefulProcessor.
+   * Function to create new or return existing single value state variable of given type with ttl.
+   * State values will not be returned past ttlDuration, and will be eventually removed from the
+   * state store. Any state update resets the ttl to current processing time plus ttlDuration.
+   *
+   * The user must ensure to call this function only within the `init()` method of the
+   * StatefulProcessor.
    *
    * @param stateName
    *   \- name of the state variable
    * @param valEncoder
    *   \- SQL encoder for state variable
+   * @param ttlConfig
+   *   \- the ttl configuration (time to live duration etc.)
    * @tparam T
    *   \- type of state variable
    * @return
    *   \- instance of ValueState of type T that can be used to store state persistently
    */
-  def getValueState[T](stateName: String, valEncoder: Encoder[T]): ValueState[T]
+  def getValueState[T](
+      stateName: String,
+      valEncoder: Encoder[T],
+      ttlConfig: TTLConfig): ValueState[T]
 
   /**
    * Function to create new or return existing single value state variable of given type with ttl.
    * State values will not be returned past ttlDuration, and will be eventually removed from the
    * state store. Any state update resets the ttl to current processing time plus ttlDuration.
    *
    * The user must ensure to call this function only within the `init()` method of the
-   * StatefulProcessor.
+   * StatefulProcessor. Note that this API uses the implicit SQL encoder in Scala.
```

Comment:
nit: we tend to add `(Scala-specific) ` at the start of method doc for Scala API. See methods in Dataset.

Reply:
Done